explorer/client/src/components/ownedobjects/views/OwnedCoinView.tsx

```diff
@@ -0,0 +1,152 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { Coin } from '@mysten/sui.js';
+import React, { useCallback, useEffect, useState } from 'react';
+
+import { ReactComponent as ContentIcon } from '../../../assets/SVGIcons/closed-content.svg';
+import { handleCoinType } from '../../../utils/stringUtils';
+import Longtext from '../../longtext/Longtext';
+import Pagination from '../../pagination/Pagination';
+import { type DataType, ITEMS_PER_PAGE } from '../OwnedObjectConstants';
+
+import styles from '../styles/OwnedCoin.module.css';
+
+export default function OwnedCoinView({ results }: { results: DataType }) {
+    const CLOSED_TYPE_STRING = '';
+
+    const [openedType, setOpenedType] = useState(CLOSED_TYPE_STRING);
+
+    const [currentPage, setCurrentPage] = useState(1);
+
+    const openThisType = useCallback(
+        (thisType: string) => () => {
+            setOpenedType(thisType);
+        },
+        []
+    );
+
+    const goBack = useCallback(() => setOpenedType(CLOSED_TYPE_STRING), []);
+
+    const uniqueTypes = Array.from(new Set(results.map(({ Type }) => Type)));
+
+    // Switching the page closes any open group:
+    useEffect(() => {
+        setOpenedType(CLOSED_TYPE_STRING);
+    }, [currentPage]);
+
+    return (
+        <>
+            <div id="groupCollection" className={styles.groupview}>
+                <div className={styles.firstrow}>
+                    <div>Type</div>
+                    <div>Objects</div>
+                    <div>Balance</div>
+                </div>
+                <div className={styles.body}>
+                    {uniqueTypes
+                        .slice(
+                            (currentPage - 1) * ITEMS_PER_PAGE,
+                            currentPage * ITEMS_PER_PAGE
+                        )
+                        .map((typeV) => {
+                            const subObjList = results.filter(
+                                ({ Type }) => Type === typeV
+                            );
+                            return (
+                                <div
```

Comment:
Not blocking but I'd really like this to be broken out into another component to keep this a little cleaner.

Reply:
Implemented